packages/react-components/theme-designer/src/components/ColorTokens/ColorTokens.tsx

```diff
@@ -1,62 +1,105 @@
+const useColorOverrideReducer = (appState: AppState, brand: BrandVariants) => {
+  const initialColorOverride: ColorOverrides = Object.fromEntries(themeNames.map(currTheme => [currTheme, {}]));
+
+  const colorOverrideReducer: (
+    state: ColorOverrides,
+    action: { type: string; colorToken?: string; newValue?: Brands },
+  ) => ColorOverrides = (state, action) => {
+    switch (action.type) {
+      case 'Add Override':
+        if (!action.colorToken || !action.newValue) {
+          return state;
+        }
+        return {
+          ...state,
+          [appState.themeLabel]: { ...state[appState.themeLabel], [action.colorToken]: action.newValue },
+        };
+      case 'Reset Overrides':
+        return { ...state, [appState.themeLabel]: {} };
+      case 'Reset Custom Overrides':
+        return { ...state, customLight: {}, customDark: {} };
+      default:
+        return state;
+    }
+  };
+  return React.useReducer(colorOverrideReducer, initialColorOverride);
+};
+
 const useStyles = makeStyles({
   root: {},
   row: {
     display: 'grid',
-    gridTemplateColumns: '1fr 1fr 1fr',
+    gridTemplateColumns: '1fr 1fr 1fr .5fr',
     alignItems: 'center',
   },
 });
 
-const brandColors: Record<string, Brands> = OverridableTokenBrandColors(teamsLightTheme, brandTeams);
+const brandColors: ColorOverrideBrands = OverridableTokenBrandColors(teamsLightTheme, brandTeams);
 
 export const ColorTokens: React.FunctionComponent<ColorTokensProps> = props => {
   const styles = useStyles();
 
-  const { brand, dispatchState } = props;
+  const { brand, appState, dispatchAppState } = props;
 
-  const [overrideList, setOverrideList] = React.useState<Partial<Theme>>({});
+  const [colorOverride, dispatchColorOverride] = useColorOverrideReducer(appState, brand);
 
-  const colorOverrideReducer: (
-    state: Record<string, Brands>,
-    action: { colorToken: string; newValue: Brands },
-  ) => Record<string, Brands> = (state, action) => {
-    const overrides = { ...state, [action.colorToken]: action.newValue };
-    setOverrideList({ ...overrideList, [action.colorToken]: brand[action.newValue] });
-    dispatchState({ type: 'Overrides', overrides: overrideList });
-    return overrides;
+  const onNewOverride = (color: string, newColor: Brands) => {
+    dispatchAppState({ type: 'Override', overrides: { [color]: brand[newColor] } });
+    dispatchColorOverride({ type: 'Add Override', colorToken: color, newValue: newColor });
   };
 
-  const [colorOverrides, dispatchColorOverrides] = React.useReducer(colorOverrideReducer, {});
+  const handleResetClick = () => {
+    dispatchAppState({ type: 'Override' });
+    dispatchColorOverride({ type: 'Reset Overrides' });
+  };
 
   return (
     <div className={props.className}>
       <div className={styles.row}>
         <Caption1>Color tokens</Caption1>
         <Caption1>Assigned values</Caption1>
         <Caption1>Usage examples</Caption1>
+        <Button size="small" onClick={handleResetClick}>
```

Comment:
this 4th column feels like a lot of wasted space for a single button. maybe it should go up next to the export button as it is an app level command